giftcard/src/main/java/com/adyen/checkout/giftcard/GiftCardViewProvider.kt

```diff
@@ -23,10 +24,12 @@ internal object GiftCardViewProvider : ViewProvider {
         defStyleAttr: Int
     ): ComponentView {
         return when (viewType) {
-            GiftCardComponentViewType -> GiftCardView(context, attrs, defStyleAttr)
+            is GiftCardComponentViewType -> GiftCardView(context, attrs, defStyleAttr)
             else -> throw IllegalArgumentException("Unsupported view type")
         }
     }
 }
 
-internal object GiftCardComponentViewType : ComponentViewType
+internal class GiftCardComponentViewType(
+    override val buttonTextResId: Int = R.string.checkout_giftcard_redeem_button
```

Comment:
Same question here as for BACS: would we ever need to pass another value here? If not, I would keep it an object